optimum/habana/peft/layer.py

```diff
@@ -31,3 +31,45 @@ def GaudiAdaloraLayerSVDLinearForward(self, x: torch.Tensor, *args: Any, **kwarg
             result += (dropout(x) @ (lora_A * lora_E).T @ lora_B.T) * (scaling / ranknum)
 
     return result
+
+
+def GaudiPolyLayerLinearForward(
+    self, x: torch.Tensor, *args: Any, task_ids: torch.Tensor = None, **kwargs: Any
+) -> torch.Tensor:
+    """
+    Copied from Linear.forward: https://github.com/huggingface/peft/blob/v0.10.0/src/peft/tuners/poly/layer.py#L135
+    The only differences are:
+    - /r equal to *(1.0/r). /r makes batch_gemm BF16 failure
```

Comment:
Should we file a SW ticket. maybe by making a small UT?

Reply:
I tried to, but failed to write a UT. seems it's related with fuse op. could R&D team debug on the optimum-habana example.
the failed log is 
[09:01:46.966319][PT_BRIDGE       ][critical][tid:1296D] /npu-stack/pytorch-integration/habana_lazy/lazy_executor.h:380:
[Rank:0] Exception in Launch thread...
Check $HABANA_LOGS/ for detailssynNodeCreateWithId failed for node: batch_gemm with synStatus 26 [Generic failure]. .
and once I disable lazy mode , it's crash in backward() instead of forward()

another new issue I found in Synapse 1.16 is, if I change per_device_eval_batch_size in peft_poly_seq2seq_with_generate.py example to 8. it's crash in evaluation (not found in 1.15).
log in synapse is 06:32:10.301238][GC                    ][critical][tid:14E][C:MultiBuffer_0] set_non_persistent_section_info.cpp:181 function: handleMultiBuffers, failed condition: (minFreeTime.is_set()), message: Multibuffer 0 has no free slot for tensor f32-3_4004565_slice_0_2_0_0_0__0__bundle_1209_4004807. lifetime end of slots: [29, 31], tensor lifetime: [23, 51]